poppy/tests/test_core.py

```diff
@@ -498,6 +498,43 @@ def test_FITSOpticalElement(tempdir='./'):
     assert foe.opd_slice == 1
     assert np.allclose(foe.opd, rect_mask*1e-9)
 
+def test_OPD_in_waves_for_FITSOpticalElement():
+    pupil_radius = 1 * u.m
+    pupil = poppy.CircularAperture(radius=pupil_radius)
+    reference_wavelength = 1 * u.um
+    single_wave_1um_lens = poppy.ThinLens(
+        name='Defocus',
+        nwaves=1,
+        reference_wavelength=reference_wavelength,
+        radius=pupil_radius
+    )
+    osys = poppy.OpticalSystem(oversample=1, npix=1024)
+    osys.add_pupil(pupil)
+    osys.add_pupil(single_wave_1um_lens)
+    osys.add_detector(0.01 * u.arcsec / u.pixel, fov_pixels=3)
+    # We have applied 1 wave of defocus at 1 um, so verify the center
+    # has lower flux than at 2 um (it should be the 'hole' of the donut)
+    psf_1um = osys.calc_psf(reference_wavelength)
+    center_pixel_value = psf_1um[0].data[1,1]
+    psf_2um = osys.calc_psf(2 * reference_wavelength)
+    assert psf_2um[0].data[1,1] > psf_1um[0].data[1,1]
+    # Now, use the single_wave_1um_lens optic to make a
+    # wavelength-independent 1 wave defocus
+    lens_as_fits = single_wave_1um_lens.to_fits(what='opd', npix=1536)
```

Comment:
Any reason for using 1536 here when the above example uses 1024? Probably not critical but it's worth trying to keep the unit tests lightweight while still being thorough. (e.g. 3 pixel FOV; but pupil sampling also drives computation time of course)

Reply:
It's a magic number to make the test pass, of course! If it's 1024, you get the optic diameter sampled across 1024 pixels, whereas the npix for the optical system is something (subtly) different. If I have it as 1024, the first half (with ThinLens) doesn't agree with the trial in the second half to numerical accuracy.